lambdas/certexpiryreminder/src/main/java/uk/gov/di/ipv/cri/drivingpermit/certreminder/handler/CertExpiryReminderHandler.java

```diff
@@ -0,0 +1,103 @@
+import uk.gov.di.ipv.cri.drivingpermit.library.config.ParameterStoreService;
+import uk.gov.di.ipv.cri.drivingpermit.library.dva.configuration.DvaConfiguration;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.security.spec.InvalidKeySpecException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.di.ipv.cri.drivingpermit.library.metrics.Definitions.CERTIFICATE_EXPIRYS;
+import static uk.gov.di.ipv.cri.drivingpermit.library.metrics.Definitions.CERTIFICATE_EXPIRY_REMINDER;
+
+public class CertExpiryReminderHandler implements RequestHandler<Object, Object> {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private final DvaConfiguration dvaConfiguration;
+    private final EventProbe eventProbe;
+
+    @ExcludeFromGeneratedCoverageReport
+    public CertExpiryReminderHandler()
+            throws CertificateException, NoSuchAlgorithmException, InvalidKeySpecException {
+        ParameterStoreService parameterStoreService =
+                new ParameterStoreService((ParamManager.getSsmProvider()));
+
+        dvaConfiguration = new DvaConfiguration(parameterStoreService);
+
+        this.eventProbe = new EventProbe();
+    }
+
+    public CertExpiryReminderHandler(DvaConfiguration dvaConfiguration, EventProbe eventProbe) {
+        LOGGER.info("CONSTRUCTING...");
+        this.dvaConfiguration = dvaConfiguration;
+        this.eventProbe = eventProbe;
+    }
+
+    @Override
+    @Metrics(captureColdStart = true)
+    @Logging(correlationIdPath = CorrelationIdPathConstants.EVENT_BRIDGE)
+    public Object handleRequest(Object input, Context context) {
+        LOGGER.info("Handling requests");
+
+        Map<String, LocalDate> certificates = new HashMap<String, LocalDate>();
+
+        for (Map.Entry<String, X509Certificate> certificate :
+                dvaConfiguration.getDVACertificates().entrySet()) {
+            LOGGER.info("Loading Certificates...");
+            Date date = certificate.getValue().getNotAfter();
+            certificates.put(certificate.getKey(), convertToLocalDate(date));
+        }
+
+        LOGGER.info("Setting expiry window");
+        LocalDate expiryWindow = LocalDate.now().plusWeeks(4);
+        Map<String, String> certExpiryMap = new HashMap<String, String>();
+        List<String> expiredCertsList = new ArrayList<String>();
+
+        for (Map.Entry<String, LocalDate> certificate : certificates.entrySet()) {
+            LOGGER.info("Checking Certificates...");
+
+            if (certificate.getValue().isAfter(LocalDate.now())
+                    && certificate.getValue().isBefore(expiryWindow)) {
+                certExpiryMap.put(certificate.getKey(), certificate.getValue().toString());
+                LOGGER.warn(
```

Comment:
Could we have an info log line above the if so we log all the certs or an else with an info log in it